### ISSUE TITLE
Emacs + HLS

### ISSUE THREAD

**Issue by arademaker** [haskell/haskell-language-server#3114]
1. MacOS with ghcup from where I installed stack, hls
2. Emacs

When I visit a .hs file inside a folder created with stack, the hls starts in the parent folder of the project folder.. I believe this forces the two errors in the screenshot:

1. the wrong advice to change the module name
2. the suggestion of watch the parent folder files3. 

Where is the error? In the HLS-server? In the emacs packages? 

<img width="1237" alt="image" src="https://user-images.githubusercontent.com/86403/185530254-ddaadfca-0c09-448a-8a2d-c632b35a8ecd.png">

**Comment by ParetoOptimalDev** [comment]
Try `M-x customize-variable lsp-file-watch-ignored` and adding your largest folders within `/Users/ar/Temp` that aren't Haskell source files you want to watch to the ignore list.

The issue for this seems to be https://github.com/emacs-lsp/lsp-mode/issues/713.

**Comment by michaelpj** [comment]
> When I visit a .hs file inside a folder created with stack, the hls starts in the parent folder of the project folder

When you open a file with `lsp-mode` like this for the first time, it should ask you where the project root is. Perhaps you gave it the wrong answer?

https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/#check-project-roots

**Comment by arademaker** [comment]
But the minor problem here is the file watcher. The real problem is that the root folder is not correctly initialized and not even the right version of the Haskell-language-server. Is I call the Haskell-language-server-wrapper in the command line in the project root folder, it works fine.

**Comment by michaelpj** [comment]
@arademaker did you see my comment? what did you set the project root to in lsp-mode?

**Comment by arademaker** [comment]
hum, I didn't know about the `lsp-describe-session`. The screenshot is below

I don't know why the other sessions are showing up here, two of them are old http://leanprover.github.io working folders... Why are they here if I don't have open buffers? The lsp-mode is NOT asking me about the project root folder..



<img width="1387" alt="image" src="https://user-images.githubusercontent.com/86403/185754368-21086844-471a-4ce0-b39d-a4e751790e53.png">

**Comment by arademaker** [comment]
Hi @michaelpj , note that the file I am opening is Main.hs in the `app` folder of the `roma` folder, the actual root folder of the project. This is at the container folder `Temp`

```
  /Users/ar/Temp/roma/app:
  total used in directory 16 available 64 GiB
  drwxr-xr-x   4 ar  staff   128B Aug 19 17:58 .
  drwxr-xr-x  16 ar  staff   512B Aug 19 17:57 ..
  -rw-------   1 ar  staff    71B Aug 19 17:58 Main.hs
  -rw-------   1 ar  staff    68B Aug 19 17:55 Main.hs.~1~
```

Totally craize, don't you think?

**Comment by arademaker** [comment]
Oh... @michaelpj, you gave me some exciting clues... I followed the definition of `lsp-describe-session` and `lsp-session` and ended up finding 

```
(defcustom lsp-session-file (expand-file-name (locate-user-emacs-file ".lsp-session-v1"))
  "File where session information is stored."
  :group 'lsp-mode
  :type 'file)
```

Indeed, I had an old `.lsp-session-v1` in my `.emacs.d` folder... I removed it and things start to work as expected. I closed emacs and when I visit the Main.hs again, lsp-mode asked me about the root folder. He made a wrong guess, but with `I` option I pointed it to the `Temp/roma` folder as the root.
